src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -1042,4 +1069,162 @@ private sealed class ProjectContainerResource(ProjectResource pr) : ContainerRes
+                    },
+                    ctx.CancellationToken).ConfigureAwait(false);
+            },
+            Tags = [WellKnownPipelineTags.BuildCompute]
+        };
+
+    private static async Task CopyContainerFilesToProjectAsync(
+        IResource resource,
+        IEnumerable<ContainerFilesDestinationAnnotation> containerFilesAnnotations,
+        IServiceProvider services,
+        CancellationToken cancellationToken)
+    {
+        var logger = services.GetRequiredService<ILoggerFactory>().CreateLogger(typeof(ProjectResourceBuilderExtensions));
+        var projectMetadata = resource.TryGetLastAnnotation<IProjectMetadata>(out var metadata) ? metadata : null;
+
+        if (projectMetadata == null)
+        {
+            logger.LogWarning("Project metadata not found for resource {ResourceName}. Cannot copy container files.", resource.Name);
+            return;
+        }
+
+        var projectPath = projectMetadata.ProjectPath;
+        var publishDir = Path.Combine(Path.GetDirectoryName(projectPath)!, "bin", "Release", "publish");
+
+        // Ensure the publish directory exists
+        Directory.CreateDirectory(publishDir);
+
+        foreach (var containerFileDestination in containerFilesAnnotations)
+        {
+            var source = containerFileDestination.Source;
+
+            // Get the image name from the source resource
+            if (!source.TryGetContainerImageName(out var imageName))
+            {
+                logger.LogWarning("Cannot copy container files from {SourceName}: Source resource does not have a container image name.", source.Name);
+                continue;
+            }
+
+            logger.LogInformation("Copying container files from {ImageName} to {PublishDir}", imageName, publishDir);
+
+            // For each ContainerFilesSourceAnnotation on the source resource, copy the files
+            foreach (var containerFilesSource in source.Annotations.OfType<ContainerFilesSourceAnnotation>())
+            {
+                var sourcePath = containerFilesSource.SourcePath;
+                var destinationPath = containerFileDestination.DestinationPath;
+
+                // If destination path is relative, make it relative to the publish directory
+                if (!Path.IsPathRooted(destinationPath))
+                {
+                    destinationPath = Path.Combine(publishDir, destinationPath);
+                }
+
+                // Ensure the destination directory exists
+                Directory.CreateDirectory(Path.GetDirectoryName(destinationPath)!);
+
+                try
+                {
+                    // Create a temporary container from the image
+                    var containerName = $"temp-{resource.Name}-{Guid.NewGuid():N}";
+                    
+                    logger.LogDebug("Creating temporary container {ContainerName} from image {ImageName}", containerName, imageName);
+                    var createSpec = new ProcessSpec("docker")
```

Comment:
Don't call into docker directly. Instead refactor all this code to be a new method on the IContainerRuntime interface called "CopyContainerFiles". With implementations for both `DockerContainerRuntime` and `PodmanContainerRuntime`.

Then get an IContainerRuntime instance like the following code:

https://github.com/dotnet/aspire/blob/d13957398c17ea21629f9a0f9c493d29d27691e6/src/Aspire.Hosting/Publishing/ResourceContainerImageBuilder.cs#L139-L144